File: two_sum.rb
# 1
# Brute force, Space: O(1) Time: O(N^2)
def two_sum(nums, target)
    l = nums.length

    l.times do |outer|
        inner = 0
        while inner < outer
            return [inner, outer] if nums[inner] + nums[outer] == target
            inner += 1
        end
    end
end

File: two_sum.js
// 1
// anchor/runner pointer solution
// Time: O(N^2) Space: O(1)
var twoSum = function(nums, target) {
  var result;

  for (var i = 0; i < nums.length; i += 1) {
    result = iterateCheck(nums, i, target);
    if (result) {
      break;
    }
  }

  return result;
};

var iterateCheck = function(nums, start, target) {
  for (var j = start + 1; j < nums.length; j += 1) {
    if (nums[start] + nums[j] === target) {
      return [start, j];
    }
  }
};
